injector/mapping.go

```diff
@@ -0,0 +1,143 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package injector
+
+import (
+	"context"
+	"fmt"
+
+	servicebindingv1alpha3 "github.com/servicebinding/service-binding-controller/apis/v1alpha3"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+// DefaultWorkloadMapping applies default values to a workload mapping. The default values are appropriate for a
+// PodSpecable resource. A deep copy of the workload mapping is made before applying the defaults.
+func DefaultWorkloadMapping(mapping *servicebindingv1alpha3.ClusterWorkloadResourceMappingTemplate) *servicebindingv1alpha3.ClusterWorkloadResourceMappingTemplate {
```

Comment:
would you consider making receiver function?

```
func (mapping **servicebindingv1alpha3.ClusterWorkloadResourceMappingTemplate)
```

Reply:
this method should end up as ClusterWorkloadResourceMappingTemplate#Default, whether or not we actually install a mutating webhook. I want to keep this separate for the moment to avoid conflicts with the webhooks pr.

Reply:
I setup webhooks for the ClusterWorkloadResoruceMapping and move the default logic there. This will conflict with #51. I'll rebase whichever ends up conflicted.